flashlight/ext/image/fl/models/ViT.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "flashlight/ext/image/fl/models/ViT.h"
+
+namespace fl {
+namespace ext {
+namespace image {
+
+ViT::ViT(
+    const int nLayers,
+    const int hiddenEmbSize,
+    const int mlpSize,
+    const int nHeads,
+    const float pDropout,
+    const float pLayerDrop,
+    const int nClasses)
+    : nLayers_(nLayers),
+      hiddenEmbSize_(hiddenEmbSize),
+      mlpSize_(mlpSize),
+      nHeads_(nHeads),
+      pDropout_(pDropout),
+      patchEmbedding_(
+          std::make_shared<Conv2D>(3, hiddenEmbSize_, 16, 16, 16, 16)) {
+  // Class token
+  params_.emplace_back(VisionTransformer::initLinear(1, hiddenEmbSize_));
+  // af::normal(af::dim4(hiddenEmbSize_, 1, 1, 1), 0.2), true);
+
+  // Positional embedding
+  // params_.emplace_back(
+  //     VisionTransformer::initLinear(14 * 14 + 1, hiddenEmbSize_));
+  params_.emplace_back(
+      af::normal(af::dim4(hiddenEmbSize_, 14 * 14 + 1, 1, 1), 0.2), true);
```

Comment:
stddev is 0.02 https://github.com/google-research/vision_transformer/blob/master/vit_jax/models.py#L179